Assets/Scripts/Enviroment/EnvironmentSpawner.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Unity.Mathematics;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+public class EnvironmentSpawner : MonoBehaviour
+{
+    [Header("Spawnable Items:")]
+    public GameObject[] environmentParts;
+    public bool areTrees;
+    
+    [Header("Spawn Constraints:")]
+    public float spawnAreaXMin;
+    public float spawnAreaXMax;
+    public float spawnAreaY;
+    public float spawnCooldown;
+    private float currentSpawnCooldown;
+    
+    [Header("Random Rotation:")]
+    public bool randomRotation;
+    public float randomRotationMin;
+    public float randomRotationMax;
+    
+    [Header("Random Scale:")]
+    public bool randomScale;
+    public float randomScaleMin;
+    public float randomScaleMax;
+    
+    // Update is called once per frame
+    void Update()
+    {
+        Spawn();
+    }
+
+    void Spawn()
+    {
+        currentSpawnCooldown -= Time.deltaTime;
+
+        if (currentSpawnCooldown <= 0)
+        {
+            var rotation = 0f;
+            var scale = 1f;
+            
+            float randomSpawnPos = Random.Range(spawnAreaXMin, spawnAreaXMax);
+            float randomRot = Random.Range(randomRotationMin, randomRotationMax);
+            float randomScaleFactor = Random.Range(randomScaleMin, randomScaleMax);
+            
+            int randomSpawnNR = Random.Range(0, environmentParts.Length-1);
+            
+            if (randomRotation)
+            {
+                rotation = randomRot;
+            }
+
+            if (randomScale)
+            {
+                scale = randomScaleFactor;
+            }
+            
+            
+            var objectMade = Instantiate(environmentParts[randomSpawnNR], new Vector2(randomSpawnPos, spawnAreaY), Quaternion.Euler(new Vector3(0, 0, rotation)));
+
+            objectMade.transform.localScale = new Vector3(scale, scale, scale);
+            
+            currentSpawnCooldown = spawnCooldown;
+        }
+    }
```

Comment:
```suggestion
    void Update()
    {
        currentSpawnCooldown -= Time.deltaTime;

        if (currentSpawnCooldown <= 0)
        {
            Spawn();
        }    
    }

    void Spawn()
    {
            var rotation = 0f;
            var scale = 1f;
            
            float randomSpawnPos = Random.Range(spawnAreaXMin, spawnAreaXMax);
            float randomRot = Random.Range(randomRotationMin, randomRotationMax);
            float randomScaleFactor = Random.Range(randomScaleMin, randomScaleMax);
            
            int randomSpawnNR = Random.Range(0, environmentParts.Length-1);
            
            if (randomRotation)
            {
                rotation = randomRot;
            }

            if (randomScale)
            {
                scale = randomScaleFactor;
            }
            
            
            var objectMade = Instantiate(environmentParts[randomSpawnNR], new Vector2(randomSpawnPos, spawnAreaY), Quaternion.Euler(new Vector3(0, 0, rotation)));

            objectMade.transform.localScale = new Vector3(scale, scale, scale);
            
            currentSpawnCooldown = spawnCooldown;
    }
```

Je spawn functie deedt niet alleen het spawnenen. Je mag nog steeds dingen doen in de update, maar nu heb je letterlijk het gene wat je zou doen in de update verplaatst naar een functie.

Reply:
dit had ik eerst en dacht dat het andere beter was, maar ik zet het wel terug dan